src/main/java/net/fabricmc/fabric/mixin/entity/MixinLivingEntity.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.entity;
+
+import net.fabricmc.fabric.block.Climbable;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.LivingEntity;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+@Mixin(LivingEntity.class)
+public abstract class MixinLivingEntity {
+
+
+    @Inject(method = "canClimb", at = @At(value = "RETURN", ordinal = 1), locals = LocalCapture.CAPTURE_FAILHARD, cancellable = true)
+    public void canClimb(CallbackInfoReturnable<Boolean> cir, final BlockState state, final Block block) {
+        if (block instanceof Climbable) {
+            cir.setReturnValue(((Climbable) block).canClimb((LivingEntity) (Object) this, state, ((LivingEntity) (Object) this).getPos()));
```

Comment:
The self-cast to LivingEntity should be stored in a local variable for these two references.

Reply:
Done.